Trouble writing tests when using OpenShift Routes

## Type of question

I'm trying to build a test suite for a controller for OpenShift Routes

## Question

**What did you do?**
I followed the [Test Framework Documentation](https://github.com/operator-framework/operator-sdk/blob/master/doc/test-framework/writing-e2e-tests.md) and the [memcached sample](https://github.com/operator-framework/operator-sdk-samples/tree/master/memcached-operator) to try to build a test framework for my Route controller. Here is the main test method:

```
func TestCertCtrl(t *testing.T) {
	routeList := &routev1.RouteList{
		TypeMeta: metav1.TypeMeta{
			Kind:       "Route",
			APIVersion: "route.openshift.io/v1",
		},
	}
	err := framework.AddToFrameworkScheme(routev1.AddToScheme, routeList)
	if err != nil {
		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
	}
	// run subtests
	t.Run("test-group", func(t *testing.T) {
		t.Run("Cluster", SetupCluster)
		t.Run("Cluster2", SetupCluster)
	})
}
```

Then I run the test with:
```
operator-sdk test local ./test/e2e 
```

**What did you expect to see?**
I expected the single test I've written to run successfully.

**What did you see instead? Under which circumstances?**

Got the following error:
```
INFO[0000] Testing operator locally.                    
--- FAIL: TestCertCtrl (1.73s)
    --- FAIL: TestCertCtrl/test-group (0.00s)
        --- FAIL: TestCertCtrl/test-group/Cluster (0.19s)
        	client.go:57: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster-1552959958/cert-operator) created
        	cert_test.go:78: failed to initialize cluster resources: failed to unmarshal object spec: (Object 'Kind' is missing in 'null')
        	client.go:75: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster-1552959958/cert-operator) successfully deleted
        --- FAIL: TestCertCtrl/test-group/Cluster2 (0.21s)
        	client.go:57: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster2-1552959958/cert-operator) created
        	cert_test.go:78: failed to initialize cluster resources: failed to unmarshal object spec: (Object 'Kind' is missing in 'null')
        	client.go:75: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster2-1552959958/cert-operator) successfully deleted
```

**Environment**
* operator-sdk version:

```
$ git describe
v0.5.0
```

* Kubernetes version information:

 ```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2018-10-10T16:38:01Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0+d4cacc0", GitCommit:"d4cacc0", GitTreeState:"clean", BuildDate:"2019-01-14T00:24:11Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"linux/amd64"}
```

* Kubernetes cluster kind: 

Embedded test framework cluster with Route object added to API scheme.

**Additional context**
Add any other context about the question here.

@hasbro17 I just tried replacing the existing operator.yaml with a new, simpler one that follows the conventions:

```
$ cat deploy/operator.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: cert-operator
  labels:
    app: cert-operator
spec:
  replicas: 3
  selector:
    matchLabels:
      app: cert-operator
      deployment: cert-operator
  template:
    metadata:
      labels:
        app: cert-operator
        deployment: cert-operator
    spec:
      containers:
        - name: cert-operator
          image: cert-operator:latest
          ports:
          - containerPort: 60000
            name: metrics
          imagePullPolicy: Always
          env:
            - name: WATCH_NAMESPACE
              valueFrom:
                fieldRef:
                  fieldPath: metadata.namespace
            - name: OPERATOR_NAME
              value: "cert-operator"
```

The result looks to be the same:
```
$ operator-sdk test local ./test/e2e/
INFO[0000] Testing operator locally.                    
--- FAIL: TestCertCtrl (1.66s)
    --- FAIL: TestCertCtrl/test-group (0.00s)
        --- FAIL: TestCertCtrl/test-group/Cluster (0.27s)
        	client.go:57: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster-1553606067/cert-operator) created
        	cert_test.go:78: failed to initialize cluster resources: failed to unmarshal object spec: (Object 'Kind' is missing in 'null')
        	client.go:75: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster-1553606067/cert-operator) successfully deleted
        --- FAIL: TestCertCtrl/test-group/Cluster2 (0.27s)
        	client.go:57: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster2-1553606067/cert-operator) created
        	cert_test.go:78: failed to initialize cluster resources: failed to unmarshal object spec: (Object 'Kind' is missing in 'null')
        	client.go:75: resource type ServiceAccount with namespace/name (certctrl-test-group-cluster2-1553606067/cert-operator) successfully deleted
FAIL
FAIL	github.com/redhat-cop/cert-operator/test/e2e	3.309s
Error: failed to exec []string{"go", "test", "./test/e2e//...", "-namespacedMan", "/tmp/namespaced-manifest.yaml538847443", "-globalMan", "/tmp/global-manifest.yaml738572822", "-root", "/home/esauer/go/src/github.com/redhat-cop/cert-operator"}: exit status 1
Usage:
  operator-sdk test local <path to tests directory> [flags]
```

--- a/internal/util/yamlutil/scan.go
+++ b/internal/util/yamlutil/scan.go
@@ -65,7 +65,7 @@ func (s *Scanner) Scan() bool {
 			s.err = err
 			return false
 		}
-		if tok == nil {
+		if len(bytes.TrimSpace(tok)) == 0 {
 			s.empties++
 			if s.empties > maxExecutiveEmpties {
 				panic("yaml.Scan: too many empty tokens without progressing")

